e2e/spintainer_test.go

```diff
@@ -88,7 +88,8 @@ func newSpintainerExecutor(namespace string) *spinapps_v1alpha1.SpinAppExecutor
 		Spec: spinapps_v1alpha1.SpinAppExecutorSpec{
 			CreateDeployment: true,
 			DeploymentConfig: &spinapps_v1alpha1.ExecutorDeploymentConfig{
-				SpinImage: generics.Ptr("ghcr.io/fermyon/spin:v2.7.0"),
+				// TODO: pin this to Spin 3.0 once released
+				SpinImage: generics.Ptr("ghcr.io/fermyon/spin:canary"),
```

Comment:
Will this break if canary introduces breaking changes? Should this be pointed at a SHA so we don't accidentally break in the future?

Reply:
We don't produce sha-tagged Spin images otherwise i would've gone with that too https://github.com/fermyon/spin/pkgs/container/spin

Reply:
Ah, I see.